pycbc/events/coinc_rate.py

```diff
@@ -0,0 +1,156 @@
+#
+# =============================================================================
+#
+#                                   Preamble
+#
+# =============================================================================
+#
+""" This modules contains functions for calculating and manipulating
+coincident triggers.
+"""
+
+import itertools
+import numpy
+import pycbc.detector
+
+
+def multiifo_noise_coinc_rate(rates, ifos, slop):
+    """
+    Calculate the expected rate of noise coincidences for multiple detectors
+
+    Parameters
+    ----------
+    rates: list of lists/arrays
+        List of vector of single-detector trigger rates
+        in order of template id
+    ifos: list of strings
+        list of ifos in e.g. 'H1' form
+    slop: float
+        time to be added to time-of-flights between detectors to account
+        for timing error
+
+    Returns
+    -------
+    expected_coinc_rates: dict
+        A dictionary of ifo combined rates keyed on the combination string
+        value is the expected coincidence rate in the combination given the
+        individual detector trigger rates
+    """
+    ifos = numpy.array(ifos)
+    rates = numpy.array(rates)
+    expected_coinc_rates = {}
+    n_ifos = len(ifos)
+    assert len(rates) == n_ifos
+    # Calculate coincidence for all-ifo combination
+    # multiply the two rates and by the overlap time
+    allowed_area = multiifo_noise_coincident_area(ifos, slop)
+    rateprod = [numpy.prod(rs) for rs in zip(*rates)]
+    ifostring = ''.join(ifos)
```

Comment:
in other places i think we are using a different layout for sets of ifos, maybe want to be consistent here?

Reply:
updated